x/cronos/keeper/evmhooks.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	gravitytypes "github.com/peggyjv/gravity-bridge/module/x/gravity/types"
+
+	"github.com/crypto-org-chain/cronos/x/cronos/types"
+)
+
+var (
+	// BankSendEvent represent the signature of
+	// `event __CosmosNativeBankSend(address recipient, uint256 amount, string denom)`
+	BankSendEvent abi.Event
+
+	// GravitySendEvent represent the signature of
+	// `event __CosmosNativeGravitySend(address recipient, uint256 amount, string denom)`
+	GravitySendEvent abi.Event
+)
+
+func init() {
+	addressType, _ := abi.NewType("address", "", nil)
+	uint256Type, _ := abi.NewType("uint256", "", nil)
+	stringType, _ := abi.NewType("string", "", nil)
+	BankSendEvent = abi.NewEvent(
+		"__CosmosNativeBankSend",
+		"__CosmosNativeBankSend",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "denom",
+			Type:    stringType,
+			Indexed: false,
+		}},
+	)
+	GravitySendEvent = abi.NewEvent(
+		"__CosmosNativeGravitySend",
+		"__CosmosNativeGravitySend",
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "recipient",
+			Type:    addressType,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "amount",
+			Type:    uint256Type,
+			Indexed: false,
+		}, abi.Argument{
+			Name:    "denom",
+			Type:    stringType,
+			Indexed: false,
+		}},
+	)
+}
+
+type BankSendHook struct {
+	bankKeeper   types.BankKeeper
+	cronosKeeper Keeper
+}
+
+func NewBankSendHook(bankKeeper types.BankKeeper, cronosKeeper Keeper) *BankSendHook {
+	return &BankSendHook{
+		bankKeeper:   bankKeeper,
+		cronosKeeper: cronosKeeper,
+	}
+}
+
+func (h BankSendHook) PostTxProcessing(ctx sdk.Context, txHash common.Hash, logs []*ethtypes.Log) error {
+	for _, log := range logs {
```

Comment:
could be cleanup a bit to loop only one time

I think instead of creating 2 hooks (for bank and gravity), you can define another interface which implement EventProcessing

Reply:
Yeah, do you mean to register the log processors into sth like: `event_id -> processer`? sounds good.

Reply:
Will the log/event order make a difference? And in some cases a processor may need to handle multiple events?

Reply:
I just pushed a new implementation.

> Will the log/event order make a difference? And in some cases a processor may need to handle multiple events?

Currently we only have one handler for one event, if we need to support more, can change the `EventID` method to return an array.

Reply:
where is decimal conversion 
10^8 -> 10^18 ?

Reply:
https://github.com/crypto-org-chain/cronos/pull/35
that is handled in this PR